flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -172,10 +178,9 @@ else if (barrierId > currentCheckpointId) {
 			}
 
 			releaseBlocksAndResetBarriers();
-			notifyCheckpoint(receivedBarrier, bufferedBytes, latestAlignmentDurationNanos);
-			return true;
+			notifyCheckpoint(receivedBarrier, latestAlignmentDurationNanos);
+			notifyCheckpointCompletedOrCanceled(barrierId);
```

Comment:
I am considering the call sequence for `notifyCheckpoint` and `notifyCheckpointCompletedOrCanceled`.
`notifyCheckpoint` might take some time for local IO operations I guess. If we can `notifyCheckpointCompletedOrCanceled` beforehand to unblock upstream to send data, then it might get benefit for performance.

Reply:
As suggested above, I move notify to the end of this method which unifies the notify process

Reply:
As suggested above, I have moved the notify to the end of the method to unify the notifications.